streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -98,16 +98,21 @@
     private final UUID processId;
 
     public KafkaStreams(TopologyBuilder builder, Properties props) {
-        this(builder, new StreamsConfig(props));
+        this(builder, new StreamsConfig(props), new DefaultKafkaClientSupplier());
+    }
+
+    public KafkaStreams(TopologyBuilder builder, StreamsConfig config) {
+        this(builder, config, new DefaultKafkaClientSupplier());
     }
 
     /**
      * Construct the stream instance.
      *
      * @param builder The processor topology builder specifying the computational logic
      * @param config The stream configs
+     * @param clientSupplier The kafka clients supplier which provides clients for this KafkaStream
```

Comment:
nit: "which provide underlying producer and consumer clients for this {@code KafkaStream} instance".